PROTOCOL.md

```diff
@@ -995,6 +1065,178 @@ Within the checkpoint, the `remove` struct does not contain the `stats` and `tag
+```
+
+Another example of a v2 spec checkpoint without sidecars:
+```
+{"checkpointMetadata":{"flavor":"flat","tags":{}}}
+{"metaData":{...}}
+{"protocol":{...}}
+{"txn":{"appId":"3ba13872-2d47-4e17-86a0-21afd2a22395","version":364475}}
+{"add":{"path":"date=2017-12-10/part-000...c000.gz.parquet",...}
+{"add":{"path":"date=2017-12-09/part-000...c000.gz.parquet",...}
+{"remove":{"path":"date=2017-12-08/part-000...c000.gz.parquet",...}
+```
+
+#### V1 Spec
+
+V1 Spec do not support [sidecar files](#sidecar-files) and [checkpoint metadata](#checkpoint-metadata).
+These are flat checkpoints which contains all actions mentioned [here](#checkpoints-1).
+
+### Checkpoint Naming Scheme
+Delta supports following three checkpoint naming scheme:
+
+#### UUID-named checkpoint
+This naming scheme represents a [V2 spec checkpoint](#v2-spec) with following file name: `n.checkpoint.u.{json/parquet}`,
+where `u` is a UUID and `n` is the snapshot version that this checkpoint represents.
+The UUID-named checkpoints may be in JSON or parquet format. Since these are following [V2 spec](#v2-spec), so they must
+have a [checkpoint metadata](#checkpoint-metadata) action and may reference zero or more checkpoint [sidecar files](#sidecar-files).
+
+Example-1: Json UUID-named checkpoint with sidecars
+
+```
+00000000000000000010.checkpoint.80a083e8-7026-4e79-81be-64bd76c43a11.json
+_sidecars/3a0d65cd-4056-49b8-937b-95f9e3ee90e5.parquet
+_sidecars/016ae953-37a9-438e-8683-9a9a4a79a395.parquet
+_sidecars/7d17ac10-5cc3-401b-bd1a-9c82dd2ea032.parquet
+```
+
+Example-2: Parquet UUID-named checkpoint with sidecars
+
+```
+00000000000000000020.checkpoint.80a083e8-7026-4e79-81be-64bd76c43a11.parquet
+_sidecars/3a0d65cd-4056-49b8-937b-95f9e3ee90e5.parquet
+_sidecars/016ae953-37a9-438e-8683-9a9a4a79a395.parquet
+```
+
+Example-3: Json UUID-named checkpoint without sidecars
+
+```
+00000000000000000112.checkpoint.80a083e8-7026-4e79-81be-64bd76c43a11.json
+```
+
+#### Classic checkpoint
+
+A classic checkpoint for version `n` uses the file name `n.checkpoint.parquet`. For example:
+
+```
+00000000000000000010.checkpoint.parquet
+```
+
+If two checkpoint writers race to create the same classic checkpoint, the latest writer wins.
+However, this should not matter because both checkpoints should contain the same information and a
+reader could safely use either one.
+
+A classic checkpoint could:
+1. Either follow [V1 spec](#v1-spec) or
+2. Could follow [V2 spec](#v2-spec). This is possible only when
+[V2 Checkpoint table feature](#v2-checkpoint-table-feature) is enabled. In this case it must include
+[checkpoint metadata](#checkpoint-metadata) and may or may not have [sidecar files](#sidecar-file-information).
+
+#### Multi-part checkpoint
+Multi-part checkpoint uses parquet format.
+This checkpoint type is [deprecated](#problems-with-multi-part-checkpoints) and writers should avoid using it.
```

Comment:
Mention here that writers are forbidden to use it when v2 checkpoints are enabled?

Reply:
This is mentioned 4-5 lines below. Also mentioned in "V2 Checkpoints table feature" section.